### ISSUE TITLE
Incorrect Africa/Casablanca offset

### ISSUE THREAD

**Issue by trzewiczek** [spencermountain/spacetime#400]
I've run into a problem with `Africa/Casablanca` timezone. It presents itself as it was `UTC+0:00` while for the given date it should be `UTC+1:00`. 

Please, see example session:

```javascript
> spacetime.version
'7.4.7'

> spacetime.now('utc').format('{numeric} {time}')
'2023/10/12 9:04am'

> spacetime.now('Africa/Casablanca').format('{numeric} {time}')
'2023/10/12 9:04am'

> spacetime.now('Africa/Casablanca').timezone()
{
  name: 'Africa/Casablanca',
  hasDst: true,
  default_offset: 1,
  hemisphere: 'North',
  current: { offset: 0, isDST: false },
  change: { start: '03/19:03', back: '04/23:02' }
}
```

The timezone according to Duckduckgo is:
```
10:04 WEST (UTC+1)
Casablanca, Morocco • Thursday, 12 October 2023
```

The issue came to light while working with both `spacetime` and `dayjs` (in which `Africa/Casablanca` is `UTC+1:00`) in one code base.

**Comment by simplypixi** [comment]
Same issue here :(

**Comment by spencermountain** [comment]
okay, i've got a fix for `africa/casablanca` and `africa/el_aaiun` for this reversey-dst thing that we somehow never noticed until now.

The change involves setting these 2 iana codes as southern hemisphere, which they're not - but given that it [drops an hour](https://www.timeanddate.com/time/zone/morocco/casablanca), midway through the year, that's how it'll get modelled internally.

If anyone can find other IANA codes that do this, but are in the northern hemisphere, I'd love to know. There's clearly a hole in the tests for this.

released as `7.4.8`. let me know if you see any more issues.
cheers
